### ISSUE TITLE
[Bug][GitHub] github-Pull Request issue no data

### ISSUE THREAD

**Issue by SxjAccount** [apache/devlake#7255]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-devlake/issues?q=is%3Aissue) and found no similar issues.

### What happened

When pulling the detailed information of pull requests, the information of pull_request_issues (the regularization of Cross Domain has already been configured) was not successfully pulled

### What do you expect to happen

The data in the pull-requests table can be retrieved from GitHub normally

### How to reproduce

![20240329155108](https://github.com/apache/incubator-devlake/assets/85595603/6b814947-7984-4b58-8dc4-b4c028ae23e3)
![20240329155124](https://github.com/apache/incubator-devlake/assets/85595603/63aa1ee1-fbba-4e3a-83a2-381744a374eb)
![20240329155130](https://github.com/apache/incubator-devlake/assets/85595603/4bfacc90-e853-4703-9088-3bb9b473e272)
![20240329155138](https://github.com/apache/incubator-devlake/assets/85595603/45c22336-ec0f-4357-9106-67d17ccae4c9)
![20240329155211](https://github.com/apache/incubator-devlake/assets/85595603/123d45e2-fef1-49d4-bd91-58f6b9508d29)

### Version

v0.20.0-beta1

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by Startrekzky** [comment]
Hi @klesh , please take a look at it.

**Comment by xamino** [comment]
Hello! I've been looking at DevLake to try it out and also have run into this issue – it seems like Github PRs aren't being ingested. I've verified that the token can pull the data via the REST API, so it isn't permissions.
![2024-06-21-144250_1236x841_scrot](https://github.com/apache/incubator-devlake/assets/1750073/9355242d-1fdb-4b92-9db4-367838fda3d9)
The project I was using to test it is private and has over 400 PRs on it, so something should be there. Checking the other tables (specifically the CICD run one) shows data existing, so this seems to me to really just be the PRs? I've checked comments which exist.

Maybe an additional hint: we don't use Github issues (and have turned that off from the scope), but there _is_ data for that repository about a lot of Github issues... but there was ever only a single one. If I click on one of the links the URL is 404... so where is that coming from? Not sure if related or a separate issue.

**Comment by klesh** [comment]
Github PRs are pulled via GraphQL by default, you may need to test that as well.
Is it Github Cloud or Enterprise?

**Comment by xamino** [comment]
I was using fine-grained access tokens and thus, as per the (excellent BTW) docs ([link](https://devlake.apache.org/docs/v1.0/Configuration/GitHub#fine-grained-personal-access-tokens)):
> Note: this token doesn't support GraphQL APIs. You have to disable Use GraphQL APIs on the connection page if you want to use it. However, this will significantly increase the data collection time.
Just to try it I turned it on and the data collection errors out.

Github cloud I believe – we don't run our own Github servers but pay for enterprise support and private repos.

EDIT: Well, fun. I turned GraphQL off again and re-ran the data collection, which works again now – and now I have data in the table... Weird. Did it pull the PR info via GraphQL before it failed? I was going to run the "failing" logs, but I guess now they succeeded.
